src/visualizer/svg.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * @description SVG functions.
+ * @author henneboy
+ */
+// Inspired by: https://www.w3.org/TR/2011/REC-SVG11-20110816/shapes.html#CircleElement
+
+(function(){
+    window.svg = {createGroupSvg, arrayBySecondIndex};
+    // Declaration of global consts, which are layout settings of the svg
+    const svgWidth = window.innerWidth / 2; // width of the svg is half the browser size
+    const svgLineSpace = svgWidth * 0.04; // the amount of space between each line, by the width of the svg
+    const svgLineWidth = svgWidth * 0.01;
+    const svgTextSize = svgLineSpace * 0.5;
+    const learningStyles = 4; // antal læringsstile
+    const svgHeight = 2 * learningStyles * svgLineSpace; // height of the svg, 4 lines, double linespace between lines, +2 for top&bottom
+
+    /**
+     * @summary Creates and appends a div element with and id to the body of the html-doc, and runs and appends the result of createGroupSvg to this div
+     * @param {Array} groupArray the group array with students
+     * @returns {HTMLElement} Html element with the graphical info of the group from the argument
+     */
+    function createGroupSvg(groupArray){
+        const colorArr = ["blue", "green", "red", "yellow", "lime", "orange", "magenta", "brown", "pink", "cyan", "purple", "hotpink", "chartreuse"];
+        const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+        svg.setAttribute("width", svgWidth);
+        svg.setAttribute("height", svgHeight);
+        svg.appendChild(createRect(0, 0, svgWidth, svgHeight)); // create a lightgrey rect that fills the whole svg to show it in browser.
+        let yValue = svgLineSpace;
+        for (let learnStyle = 0; learnStyle < learningStyles; learnStyle++) {
```

Comment:
Lav creation af en læringsstildimension til en funktion i stedet for det her gigantiske for loop

Reply:
Done